lib/ops/monitoring/influxdb.go

```diff
@@ -23,29 +23,64 @@ import (
 	"time"
 
 	"github.com/gravitational/gravity/lib/defaults"
+
 	"github.com/gravitational/roundtrip"
 	"github.com/gravitational/teleport/lib/httplib"
 	"github.com/gravitational/trace"
+	"k8s.io/client-go/kubernetes"
 )
 
 type influxDB struct {
 	*roundtrip.Client
 }
 
 // NewInfluxDB returns a new InfluxDB monitoring provider
-func NewInfluxDB() (Monitoring, error) {
+func NewInfluxDB(kubeClient *kubernetes.Clientset) (Monitoring, error) {
```

Comment:
I'd consider splitting it into a series of readable functions:
```go
// NewInfluxDB returns a new InfluxDB monitoring provider
func NewInfluxDB(kubeClient *kubernetes.Clientset) (Monitoring, error) {
	client, err := newInfluxDBClient(kubeClient.CoreV1())
	if err != nil {
		return nil, trace.Wrap(err)
	}
	return &influxDB{Client: client}, nil
}

func newInfluxDBClient(kclient corev1.CoreV1Interface) (*roundtrip.Client, error) {
	authParam, err := getInfluxDBCredentials(kclient)
	if err != nil {
		return nil, trace.Wrap(err)
	}
	client, err := roundtrip.NewClient(defaults.InfluxDBAddr(), "", authParam)
	if err != nil {
		return nil, trace.Wrap(err)
	}
	if err := verifyAuth(client); err == nil {
		return client, nil
	}
	if !trace.IsAccessDenied(err) {
		return nil, trace.Wrap(err)
	}
	return backwardsCompatibleClient()
}

func verifyAuth(client *roundtrip.Client) error {
	response, err := Get(client, client.Endpoint("query"), url.Values{"q": []string{showQuery}})
	if err != nil {
		return trace.Wrap(err)
	}
	if response.Code() == http.StatusUnauthorized {
		return trace.AccessDenied("auth failed")
	}
	return nil
}

func backwardsCompatibleClient() (*roundtrip.Client, error) {
	client, err := roundtrip.NewClient(defaults.InfluxDBAddr(), "",
		roundtrip.BasicAuth(
			defaults.InfluxDBAdminUser,
			defaults.InfluxDBAdminPassword,
		))
	if err != nil {
		return nil, trace.Wrap(err)
	}
	return client, nil
}
```

`lib/defaults/defaults.go`:
```go
// InfluxDBAddr returns the address of the InfluxDB cluster endpoint
func InfluxDBAddr() string {
	return fmt.Sprintf("http://%v:%v", InfluxDBServiceAddr, InfluxDBServicePort)
}
```

`lib/ops/monitoring/monitoring.go`:
```go
// getInfluxDBCredentials returns basic auth configuration for
// the InfluxDB administrator user
func getInfluxDBCredentials(kclient corev1.CoreV1Interface) (client roundtrip.ClientParam, err error) {
	secret, err := kclient.Secrets(defaults.MonitoringNamespace).Get(defaults.InfluxDBSecretName, metav1.GetOptions{})
	if err != nil {
		return nil, trace.Wrap(rigging.ConvertError(err))
	}
	username, ok := secret.Data["username"]
	if !ok {
		return nil, trace.NotFound("username for InfuxDB administrator not found")
	}
	password, ok := secret.Data["password"]
	if !ok {
		return nil, trace.NotFound("password for InfuxDB administrator not found")
	}
	return roundtrip.BasicAuth(string(username), string(password)), nil
}
```